CMakeLists.txt

```diff
@@ -119,6 +119,16 @@ cpp_cc_git_submodule(fmt BUILD PACKAGE fmt REQUIRED)
 if(NMODL_3RDPARTY_USE_FMT)
   set_property(TARGET fmt PROPERTY POSITION_INDEPENDENT_CODE ON)
 endif()
+if(NOT NMODL_3RDPARTY_USE_FMT
+   AND CMAKE_CXX_STANDARD IS_NEWER_THAN 11
+   AND ((CMAKE_CXX_COMPILER_ID STREQUAL "NVHPC"
+         AND COMPILER_VERSION_MAJOR LESS_EQUAL 22
+         AND COMPILER_VERSION_MINOR LESS_EQUAL 3)
+        OR CMAKE_CXX_COMPILER_ID STREQUAL "INTEL"
+       ))
+  message(FATAL_ERROR "fmt might generate issues with NVHPC <=22.3 and Intel compiler when installed
+                       with C++11 or later standard enabled")
```

Comment:
I haven't read through https://github.com/fmtlib/fmt/issues/3028 but is this apply to Intel as well? 🤔 
And what would be reasonable / robust solution here than `FATAL_ERROR`? (e.g. `fmt` should be compiled with <=C++14 flags?)

Reply:
It seems like the issue is applied to Intel compilers as well. I've tried to install NMODL with the deployed Intel compilers in BB5 without this fix and with both of them the tests fail with gibberish being printed in various cases

Reply:
I'm not sure FATAL_ERROR is unreasonable. We could try building `libfmt` internally with a different C++ standard to the rest of the build, but I guess it remains to be seen if that is robust.

Reply:
@olupton / @iomaganaris : if we are not able to build with Intel, isn't this means we are quite in broken state?
(as a user, we recommend building just with Intel or NVHPC compiler neuron including dependencies)

Reply:
IIUC (I haven't tested the Intel side) then this patch fixes that workflow..?
(The lack of complaints so far I guess means that not many people are actually doing that...)

Reply:
> (The lack of complaints so far I guess means that not many people are actually doing that...)

true - coreneuron is not enabled by default and few users explicitly enable this. And, this is a relatively recent change.

Reply:
In the end figuring out which `fmt` is problematic is pretty complex because any `fmt` before Olli's fix that is compiled with `CXX_STANDARD 17` with Intel/NVHPC compiler will fail, either as submodule or as an external package from `spack`. (Although by default our `spack` installs `fmt` with `CXX_STANDARD 11` and this is why we don't see an issue in the CI tests, etc)
For this reason maybe it's better to remove the `FATAL_ERROR` and instead print a `WARNING` message.
What do you guys think? Any better idea how to handle this situation?